src/transform/mod.rs

```diff
@@ -1,32 +1,74 @@
 use crate::core::{ColourModel, Image, ImageBase};
-use crate::transform::affine::translation;
 use ndarray::{array, prelude::*, s, Data};
 use ndarray_linalg::*;
 use num_traits::{Num, NumAssignOps};
 use std::cmp::{max, min};
+use std::fmt::Display;
 
 pub mod affine;
 
 #[derive(Clone, Copy, Eq, PartialEq, Debug, Hash)]
-pub enum Error {
-    InvalidTransformation,
-    NonInvertibleTransformation,
+pub enum TransformError {
+    InvalidTransform,
+    NonInvertibleTransform, }
+
+impl std::error::Error for TransformError {}
+
+impl Display for TransformError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self { 
+            InvalidTransform => return write!(f, "invalid transform" ),
```

Comment:
I think here you should do `Self::InvalidTransform` that might fix CI

Reply:
Took TransformerError::InvalidTransform to pass tests... hopefully that works.